hbase-it/src/test/java/org/apache/hadoop/hbase/HBaseClusterManager.java

```diff
@@ -159,6 +169,38 @@ public void execute() throws IOException {
     }
   }
 
+  /**
+   * Executes commands over SSH
+   */
+  protected class RemoteSudoShell extends Shell.ShellCommandExecutor {
```

Comment:
It might be required in some setups, but even then it should be already in place. Using sudo is nothing new, we already use it to start/stop services. Just check the original tunnel command.
I tested these in Ubuntu and CentOS machines and for me creating the new sudoer user on every machine, adding it to the sudo group (or wheel on CentOS), then pushing the key to the slaves with ssh-copy-id was enough.

Reply:
> Using sudo is nothing new

Oh, ok, great. I had just assumed that this was net-new. Not a concern if it's prior-art.